### PR TITLE
feat(dashboard): support specifying additional selector in PromQL

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Support specifying additional selector in the PromQL queried by Meta dashboard, which is useful when using a shared Prometheus.

Tested locally. 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by xxchan — APPROVED** [review]

File: src/meta/src/dashboard/prometheus.rs [hunk 1]
```diff
@@ -92,7 +93,7 @@ pub async fn list_prometheus_cluster(
             .map(PrometheusVector::from)
             .collect();
         let memory_query =
-            "avg(process_resident_memory_bytes{job=~\"compute|meta|frontend\"}) by (job,instance)";
+            format!("avg(process_resident_memory_bytes{{job=~\"compute|meta|frontend\",{}}}) by (job,instance)", srv.prometheus_selector);
```
Comment (BugenZhao) [inline src/meta/src/dashboard/prometheus.rs:96]: Is trailing comma okay?

**Review by BugenZhao — APPROVED** [review]
Looks good. Does this mean that we should prefer meta dashboard than #13422?

File: src/meta/src/dashboard/prometheus.rs (hunk 1 above)
Comment (fuyufjh) [reply src/meta/src/dashboard/prometheus.rs:96]: Yes. 

- ✅ `metric_name{}`
- ✅ `metric_name{foo="bar"}`
- ✅ `metric_name{foo="bar",}`
- ❌ `metric_name{,}`
- ❌ `metric_name{foo="bar",,}`

**Comment by fuyufjh** [comment]
> Looks good. Does this mean that we should prefer meta dashboard than #13422?

Don't know. #13422 assumes Grafana can connect to Meta, while this assumes Meta can connect to Prometheus. Honestly, either is not 100% guaranteed in different environments.

I did this because @arkbriar mentioned that "we can assume Prometheus is always reachable from Meta", but I am still a little bit suspicious. Anyway, this is not a breaking change, so I'd like to provide this to the cloud team first to see whether they can make it work.

**Comment by arkbriar** [comment]
> we can assume Prometheus is always reachable from Meta

Can't remember that xD... I think it's always correct to make prom an option.

**Comment by fuyufjh** [comment]
> @BugenZhao:
> Is it always possible to directly (and safely) access the Prometheus’ HTTP API endpoint from Kernel in cloud or operator deployments? We want to visualize the back-pressure and throughput together with the stream graph in Kernel’s dashboard.

> @arkbriar:
> Fantastic idea.
> > Is it always possible to directly (and safely) access the Prometheus’ HTTP API endpoint from Kernel in cloud or operator deployments?
>
> Not always but you can consider it possible. (edited) 

https://risingwave-labs.slack.com/archives/C03DFGV6L2W/p1700211697959939?thread_ts=1700210498.856439&cid=C03DFGV6L2W

**Comment by arkbriar** [comment]
> > @BugenZhao:
> > Is it always possible to directly (and safely) access the Prometheus’ HTTP API endpoint from Kernel in cloud or operator deployments? We want to visualize the back-pressure and throughput together with the stream graph in Kernel’s dashboard.
> 
> > @arkbriar:
> > Fantastic idea.
> > > Is it always possible to directly (and safely) access the Prometheus’ HTTP API endpoint from Kernel in cloud or operator deployments?
> > 
> > 
> > Not always but you can consider it possible. (edited)
> 
> https://risingwave-labs.slack.com/archives/C03DFGV6L2W/p1700211697959939?thread_ts=1700210498.856439&cid=C03DFGV6L2W

Thanks for reminding. I must mean it's possible but not always true.

**Pull request merged**